iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -596,6 +596,122 @@ static LogicalResult setRootConfig(
       entryPointFn, fftOp, tileSizes, DispatchLoweringPassPipeline::CPUDefault);
 }
 
+static LogicalResult setX86DefaultParallelTileSizes(
+    linalg::GenericOp genericOp, unsigned numLoops,
+    ArrayRef<int64_t> flowTileSizes, ArrayRef<int64_t> minTileSizes,
+    ArrayRef<int64_t> maxTileSizes,
+    SmallVectorImpl<int64_t> &parallelTileSizes) {
+  parallelTileSizes.append(numLoops, 0);
+  Optional<SmallVector<int64_t, 4>> staticLoopRanges =
+      genericOp.getStaticLoopRanges();
+  for (auto loopNum : llvm::seq<unsigned>(0, numLoops)) {
+    if (flowTileSizes[loopNum]) {
+      parallelTileSizes[loopNum] =
+          getMaxTileSize(0, flowTileSizes[loopNum], minTileSizes[loopNum],
+                         minTileSizes[loopNum]);
```

Comment:
Note: We don't have to fix it in this PR. This snippet is refactored from existing codes.

@MaheshRavishankar I'm a bit outdated about this. Do we intend to pass minTileSizes as `maxSize` in `getMaxTileSize` method? The maxTileSizes is not used in setting tiling sizes. Is it a bug?

Reply:
Might be a bug actually, It should probably be `minTileSizes, maxTileSizes` here.

THough a lot of this is out of my cache. I am in no way tied to anything setting here. Its just some default I came up with based on some ad-hoc thumb rules, and change it however you feel it is worth it.